server/src/main/scala/ch.epfl.scala.index.server/Api.scala

```diff
@@ -102,21 +100,31 @@ class Api(github: Github)(implicit val ec: ExecutionContext) {
     )
   }
 
-  def releases(project: Project.Reference): Future[List[Release]] = {
+  def releases(project: Project.Reference, defaultArtifact: Option[String] = None): Future[List[Release]] = {
+
+    val artifactReference = 
+      defaultArtifact.map(artifact =>
+        List(termQuery("reference.artifact", artifact))
+      ).getOrElse(List())
+
+    val references = List(
+      termQuery("reference.organization", project.organization),
+      termQuery("reference.repository", project.repository)
+    ) ::: artifactReference
+
     esClient.execute {
       search
         .in(indexName / releasesCollection)
         .query(
             nestedQuery("reference").query(
                 bool(
                     must(
-                        termQuery("reference.organization", project.organization),
-                        termQuery("reference.repository", project.repository)
+                      references
                     )
                 )
             )
         )
-        .size(5000)
+        .size(500)
```

Comment:
artifactReference will filter releases so we only have one artifact (ex: cats-core). This is why I can put a lower limit.